test/helper/sending.bash

```diff
@@ -45,57 +47,127 @@ function _send_email() {
+# the e-mail transaction was successful. If this is not desirable, use
+# `_send_email_unchecked`.
+#
+# ## Attention
+#
+# This function assumes `CONTAINER_NAME` to be properly set (to the container
+# name the command should be executed in)!
+#
+# This function will just send the email in an "asynchronous" fashion, i.e. it will
+# send the email but it will not make sure the mail queue is empty after the mail
+# has been sent.
+function _send_email() {
+  _send_email_unchecked "${@}"
+  assert_success
 }
 
 # Like `_send_email` with two major differences:
 #
 # 1. this function waits for the mail to be processed; there is no asynchronicity
-#    because filtering the logs in a synchronous way is easier and safer!
-# 2. this function prints an ID one can later filter by to check logs
+#    because filtering the logs in a synchronous way is easier and safer;
+# 2. this function takes the name of a variable and inserts ID(s) one can later
+#    filter by to check logs.
 #
 # No. 2 is especially useful in case you send more than one email in a single
 # test file and need to assert certain log entries for each mail individually.
 #
-# This function takes the same arguments as `_send_mail`.
+# The first argument has to be the name of the variable that the e-mail ID is stored
+# in. The second argument **can** be the flag `--unchecked`; if this flag is supplied,
+# the function uses `_send_email_unchecked` instead of `_send_email`. This avoids the
+# `assert_success`. Be warned though this is only required in special situations where
+# it is still possible to `grep` for the mail IDs - otherwise this function fails.
+#
+# The rest of the arguments are the same as `_send_email`.
 #
 # ## Attention
 #
 # This function assumes `CONTAINER_NAME` to be properly set (to the container
 # name the command should be executed in)!
 #
+# Moreover, if `--data <DATA>` is specified, the additional header added implicitly
+# (with `--add-header`) may get lost, so pay attention to the data having the token
+# to place additonal headers into.
+#
 # ## Safety
 #
 # This functions assumes **no concurrent sending of emails to the same container**!
 # If two clients send simultaneously, there is no guarantee the correct ID is
 # chosen. Sending more than one mail at any given point in time with this function
 # is UNDEFINED BEHAVIOR!
 function _send_email_and_get_id() {
-  [[ -v CONTAINER_NAME ]] || return 1
+  # Get the name of the variable that the ID is stored in
+  local ID_NAME="${1:?Mail ID must be set for _send_email_and_get_id}"
+  # Get a "reference" to the content of ID_NAME so we can manipulate the content
+  local -n MAIL_ID=${ID_NAME}
+  # Export the variable so everyone has access
+  # shellcheck disable=SC2163
+  export "${ID_NAME}"
+  shift 1
 
   _wait_for_empty_mail_queue_in_container
-  _send_email "${@}"
+  if [[ ${1:-} == --unchecked ]]; then
+    shift 1
+    _send_email_unchecked "${@}" --add-header "Message-Id: ${ID_NAME}"
```

Comment:
`--header` may be more appropriate given my earlier `hostname.bats` investigation comment that compared the differences between `--header` and `--add-header`.

```suggestion
    _send_email_unchecked "${@}" --header "Message-Id: ${ID_NAME}"
```